### PR TITLE
Reduce docker compose build time

### PR DESCRIPTION
**Pull request by ambuznego**
Total build time w/ new `Dockerfile` (`staging` branch): 259.2s
Total build time w/ prior `Dockerfile` (`staging` branch): 432.1s

Based the changes on the `Dockerfile` off of: https://github.com/phoenixframework/phoenix/blob/master/guides/deployment/releases.md

Tweaked it to prevent permission errors post-release

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,22 +25,40 @@ RUN mix deps.get --only $MIX_ENV
...
 
 # build assets
+COPY priv priv
+
+# Note: if your project uses a tool like https://purgecss.com/,
+# which customizes asset compilation based on what it finds in
+# your Elixir templates, you will need to move the asset compilation step
+# down so that `lib` is available.
 COPY assets assets
-RUN rm assets/webpack.config.js
```
Comment (vicmgs) [inline Dockerfile:28]: we need to keep these lines because the webpack docker config needs to replace the regular webpack config due to env vars

**Review by vicmgs — COMMENTED** [review]
wow looks like magic lolz

File: Dockerfile (hunk 1 above)
Comment (ambuznego) [reply Dockerfile:28]: brought them back 👍 check last commit

File: Dockerfile [hunk 2]
```diff
@@ -24,22 +25,43 @@ RUN mix deps.get --only $MIX_ENV
...
+FROM alpine:3.12.1 AS app
+RUN apk add --no-cache libstdc++ openssl ncurses-libs
+
+ARG MIX_ENV
+ENV USER="nobody"
+
+WORKDIR "/home/nobody/app"
+
+COPY --from=build --chown="${USER}" /app/_build/prod/rel/console ./
+RUN chown -R nobody: .
```
Comment (michaeldjeffrey) [inline Dockerfile:63]: is this still needed if you're copying as `nobody`?

Comment (ambuznego) [reply Dockerfile:63]: it builds ok without it but then I get tons of permission errors when trying to run console. was a pain to figure out but this doesn't take nearly as long as before.

**Comment by vicmgs** [comment]
i tried using this when upgrading packages but couldn't get it to work. Closing for now

**Pull request closed without merging**